### DISCUSSION TITLE
Long Term Thinking (v2?)

### DISCUSSION THREAD

**Discussion by mikk-c** [fslaborg/Graphoscope#39] [Discussions]
I have some long term thoughts that should be ignored for now, but I think are important to write down to avoid losing them.

If we accept the task-oriented philosophy I proposed in the other discussion, we could consider providing the user even more utility functions, by incorporating behind the veil some of the imported libraries.

One example is visualization. For a v2 of the library, it could be great to have a function Degree.showDistribution that simply does that: no need for the user to pipe the same several Plotly.NET commands every time they want to perform this basic task. Graphoscope can deal with that without the users having to.

This is also considering the fact that the standard thing to visualize is not the simple degree distribution, but the complement of its cumulative, which is annoying to calculate manually every time -- having also a Degree.showDistributionCCDF would be absolute gold.

This can also lead to wider adoption of the library, because F# newbies such as myself then don't have to learn Plotly.NET on top of Graphoscope and F# (in fact, I'm having troubles myself making it work).

Of course this could be user hostile, because sometimes users want to visualize or do whatever else with the actual distribution, rather than having the package maintainers decide for them. But this can be easily solved by leaving the current Degree.distribution function alone. Then people who want to use Plotly.NET can still do it.

Same thing for having a simple Graph.show function. Currently, using Cytoscape.NET to make the visualization is pretty intimidating. We can save the user the trouble.

**Comment by LibraChris** [comment]
Thank you for your input. The functions you mentioned to visualize specific metrics and graphs directly are certainly a good idea for our next release. 
You mentioned the following visualisation functions:
- Degree.showDistribution
- Degree.showDistributionCCDF
- Graph Visualisation

Are there more examples in our current library that you think would benefit from a direct visualisation function? Or do you have suggestions for future visualisation tasks that fit here? I will collect them for a milestone checklist for our next release.
